suricata.yaml.in

```diff
@@ -160,9 +160,24 @@ outputs:
             # IP/UDP/TCP length values, and other events that render the packet invalid for further processing 
             # or describe unexpected behavior on an established stream. Networks which experience high
             # occurrences of anomalies may experience packet processing degradation.
-
-            # Enable dumping of packet header
-            # packethdr: no            # enable dumping of packet header
+            #
```

Comment:
@jasonish can you have a look at these yaml additions?

Reply:
I'm a little overwhelmed by these options.  First, whats on by default? I should see that as comment here.

```packethdr``` seems to be a bit of an outlier as it toggles a parameter, not an event type.  So it probably should not be in the middle of the other event types.

I think there is over-usage of "proto". For "protodecode" I think these are for packet decoding?  But then we have "protoparser" and "protodetect" which are for app-layer?  Also, whats the difference between "parser" and "protoparser"?

As a user, I think I'd rather just have options to turn on packet level anomalies, and application-layer level anomalies. Or supporting documentation to help make that decision.

If sticking with these options I wondering if something like the following would be more clear:
```
- anomaly:
    types:
      protodecode: no
      protparser: yes
    packethdr: yes
```

Reply:
Thanks for the feedback. I like your simplification and will adopt 2 anomaly choices:
1. Packet (Events from `DEvents`  see `decode-events.c`)
1. Application level: App-layer parser specific items and any event not included above.

```
# Anomalies are reported for the following:
# 1. Packet: Values and conditions that are detected while decoding individual packets. This includes
# invalid or unexpected values for low-level protocol lengths as well as stream related events (TCP 3-way
# handshake issues, unexpected sequence number, etc).
# 2. Application level: These denote application layer specific conditions that are unexpected or invalid.
#
 # By default, anomaly logging is disabled. Choose one or both types of anomaly logging and whether to enable logging of the packet header for packet anomalies.
   # types:
   #   packet: no
   #   application-layer: no
   #packethdr: no
```

Reply:
Would it make sense to have them enabled by default, provided `- anomaly` is set.  Then you can select the types to set them to know if you like.  But if the types is just commented out, you get all?

Reply:
They can be high volume which could be handled easily enough since there'd be separate toggles for each type.

Ok ... I'll make those changes.